### ISSUE TITLE
load_forecasts() should allow for querying forecasts from multiple timezeroes at once

### ISSUE THREAD

**Issue by nickreich** [reichlab/covidHubUtils#21]
right now, load_forecasts() uses a fairly different query structure for forecast_dates/timezeroes than the native zoltr `do_zoltar_query()` does. The difference is that we load_forecasts() specifies a single date and a window, but the native zoltar queries allow for all timezeroes to be queried (timezeroes left empty) or a specific set of timezeroes to be specified. my suggestion is that we should change the specification of load_forecasts() to accept a `forecast_date` argument that can be passed directly to `do_zoltar_query()`.

**Comment by nickreich** [comment]
My proposal is to split `load_forecasts()` into two separate functions that have something like the following use-cases:
```
## load all forecasts on any forecast date for specific models, targets, locations
f1 <- load_forecasts(
    models = c("COVIDhub-ensemble", "COVIDhub-baseline", "YYG-ParamSearch", "UMass-MechBayes"),
    targets = paste(1:4, "wk ahead inc death"),
    locations = "US",
    source = "zoltar",
    types = "quantile")

## load all forecasts on any forecast date for specific models, targets, locations
f2 <- load_forecasts(
    models = c("COVIDhub-ensemble", "COVIDhub-baseline", "YYG-ParamSearch", "UMass-MechBayes"),
    targets = paste(1:20, "wk ahead inc case"),
    locations = c("MA", "TX"),
    source = "zoltar",
    types = "point")

## load forecasts made on two specific forecast dates for specified models, targets, locations
f3 <- load_forecasts(
    models = c("COVIDhub-ensemble", "COVIDhub-baseline", "YYG-ParamSearch", "UMass-MechBayes"),
    targets = paste(1:4, "wk ahead inc case"),
    forecast_dates = c("2020-06-05", "2020-10-06"),
    locations = c("MA", "TX"),
    source = "zoltar",
    types = c("point", "quantile"))

## load latest forecasts made on a specific forecast date 
## (with required window: if window is not included this is identical to load_forecast) 
## for specified models, targets, locations
f3 <- load_latest_forecasts(
    models = c("COVIDhub-ensemble", "COVIDhub-baseline"),
    targets = paste(1:4, "wk ahead inc death"),
    forecast_date = "2020-10-05", ## note only a single date
    forecast_date_window_size = 6, 
    locations = c("MA", "CA"),
    source = "zoltar",
    types = c("point", "quantile"))
```

**Comment by nickreich** [comment]
fixing the above would also fix #22
